### ISSUE TITLE
Implementation of Bilateral Guided Upsampling

### ISSUE THREAD

**Issue by jzakirov** [mahmoudnafifi/Exposure_Correction#12]
Hi, @mahmoudnafifi 

While code for this project is not released I'd like to clarify which implementation of bilateral guided upsample did you use?
I found 3 versions:
1. Code from Google [google/bgu](https://github.com/google/bgu). 
2. Code from Google's HRNet paper [google/hdrnet](https://github.com/google/hdrnet/tree/master/hdrnet/ops)
2. Version from DeepUPE repo [Jia-Research-Lab/DeepUPE](https://github.com/Jia-Research-Lab/DeepUPE/tree/master/main/ops).

All these options have their own limitations, 1 and 3 is pretty old and can't be compiled with new CUDA10+ and TF2+, while 2 is GPU only. 

So my questions is: have you written your own custom implementation or adapted one of those?

**Comment by mahmoudnafifi** [comment]
Hi @zakajd,

The hdrnet and deepupe do the upsampling in their end-to-end training. This upsampling is not a part of our solution, I used it as a post-processing step in the testing phase only. That is why I used the non-learning one [google/bgu](https://github.com/google/bgu). As this is a post-processing step and applied only in the testing phase, the CPU implementation is fine. Reimpelemnting it is on my to-do list but can't promise it.
